File: chapter3/gcd2.rb
a, b = STDIN.gets.chomp.split(" ").map(&:to_i)

answer = 1
loop do
  if a < b
    b = b % a
  else
    a = a % b
  end

  if a == 0 || b == 0
    answer = [a, b].max
    break
  end
end

puts answer

File: chapter3/gcd2.go
package main

import (
	"fmt"
)

func main() {
	var a, b int
	fmt.Scan(&a, &b)

	answer := 1
	for {
		if a < b {
			b = b % a
		} else {
			a = a % b
		}

		if a == 0 || b == 0 {
			if a > b {
				answer = a
			} else {
				answer = b
			}
			break
		}
	}

	fmt.Println(answer)
}
